### ISSUE TITLE
hacked version of kraken-lib for ubuntu support on AWS

### ISSUE THREAD

**Issue by coffeepac** [samsung-cnct/kraken-lib#1004]
there is a short-term need for getting kraken running on aws on ubuntu.  This will result in a PR that has DO-NOT-MERGE and some instructions for how to use.  the imediate use-case is a platform for software defined storage systems.

**Comment by coffeepac** [comment]
WIP branch:  https://github.com/coffeepac/kraken-lib/tree/ubuntu-hack

**Comment by coffeepac** [comment]
going whole hog on this branch never being merged:  adding some docker-install specifics.

**Comment by coffeepac** [comment]
okay, back to where I was last week:  node name discrepencies.

**Comment by coffeepac** [comment]
was able to work around the node name discrepancy by changing the %H value in both the kraken-ssl unit file and the kubelet unit file to be the full hostname `hostname -f`.  will now try and do that in the unit file.

**Comment by coffeepac** [comment]
attempted a reboot of the node and it did not fix the glitch.  I am feeling very confident that the aws cloudprovider is overriding the hostname.

**Comment by coffeepac** [comment]
trying to tackle the docker startup step:
- when adding docker.service to existing `runcmd` for restarting services there was a bad race condition
- when I added it to its own runcmd, kubelet didn't come up but manually issuing `systemctl start kubelet` resolved the issue
- next attempt is to add kubelet to the solo docker restart runcmd.   I think its a timing issue.

**Comment by coffeepac** [comment]
Node registration issue:
- issue is that systemd and the aws cloud-provider disagree on what the host's name is.  systemd uses the short name (ip-10-x-x-x) and the cloud provider uses the fqdn (ip-10-x-x-x.internal.ec2).  these values need to agree
- if I manually set the value in the kraken-ssl unit file to be the fqdn, the cluster forms correctly
  - I am unable, so far, to figure out how to get systemd to use the fqdn in the unit file
- I am unable to tell kubelet to use anything but the fqdn, aws cloud provider seems to be hardcoding this value although I haven't gone and checked the code.  I feel like I checked this ages ago
- I attempted to remove the NodeRestriction admission controller and api-server wouldn't start.  I couldn't find any logs as the docker container was cycling really fast but I am going to back off from this for now

Next step:  update the networking container code to provide the fqdn as an environment variable and use that in the kraken-ssl unit file
